src/sql/workbench/services/fileBrowser/browser/urlBrowserDialog.ts

```diff
@@ -0,0 +1,477 @@
+		expandPath: string,
+		fileFilters: [{ label: string, filters: string[] }],
+		fileValidationServiceType: string,
+	): void {
+		this._ownerUri = ownerUri;
+		this.enableOkButton();
+		this.enableCreateCredentialsButton();
+		this.spinner = true;
+		this.show();
+		this.spinner = false;
+	}
+
+	/* enter key */
+	protected override onAccept() {
+		let selectedValue = this._sasInputBox.value;
+		if (this._okButton.enabled === true && selectedValue !== '') {
+			this.ok();
+		}
+	}
+
+
+	private enableOkButton() {
+		if (strings.isFalsyOrWhitespace(this._blobContainerSelectorBox.value) || strings.isFalsyOrWhitespace(this._sasInputBox.value) || (this._restoreDialog && strings.isFalsyOrWhitespace(this._blobContainerSelectorBox.value)) || (!this._restoreDialog && strings.isFalsyOrWhitespace(this._backupFileInputBox.value))) {
+			this._okButton.enabled = false;
+		} else {
+			this._okButton.enabled = true;
+		}
+	}
+
+	private enableCreateCredentialsButton() {
+		if (strings.isFalsyOrWhitespace(this._blobContainerSelectorBox.label)) {
+			this._sasButton.enabled = false;
+		} else {
+			this._sasButton.enabled = true;
+		}
+	}
+
+	private ok() {
+		let returnValue = '';
+		if (this._restoreDialog) {
+			returnValue = `https://${this._storageAccountSelectorBox.value}.blob${this._selectedAccount.properties.providerSettings.settings.azureStorageResource.endpointSuffix}/${this._blobContainerSelectorBox.value}/${this._backupFileSelectorBox.value}`;
+		} else {
+			returnValue = `https://${this._storageAccountSelectorBox.value}.blob${this._selectedAccount.properties.providerSettings.settings.azureStorageResource.endpointSuffix}/${this._blobContainerSelectorBox.value}/${this._backupFileInputBox.value}`;
+		}
+		this._onOk.fire(returnValue);
+		this.close('ok');
+	}
+
+
+	private close(hideReason: HideReason = 'close'): void {
+		this._onOk.dispose();
+		this.hide(hideReason);
+	}
+
+	private generateSharedAccessSignature() {
+		this.spinner = true;
+		const blobContainerUri = `https://${this._storageAccountSelectorBox.value}.blob${this._selectedAccount.properties.providerSettings.settings.azureStorageResource.endpointSuffix}/${this._blobContainerSelectorBox.value}`;
+		this._azureAccountService.getStorageAccountAccessKey(this._selectedAccount, this._selectedSubscription, this._selectedStorageAccount)
+			.then(getStorageAccountAccessKeyResult => {
+				const key1 = getStorageAccountAccessKeyResult.keyName1;
+				this._blobService.createSas(this._ownerUri, blobContainerUri, key1, this._selectedStorageAccount.name, this.nextYear())
+					.then(
+						result => {
+							const sas = result.sharedAccessSignature;
+							this._sasInputBox.value = sas;
+							this.spinner = false;
+						});
+			})
+			.catch(error => {
+				this.spinner = false;
+			});
+
+	}
+
+	private nextYear(): string {
+		const today = new Date();
+		const nextYear = new Date(today.getFullYear() + 1, today.getMonth(), today.getDay());
+		return nextYear.toLocaleString().split(',')[0];
```

Comment:
Is calling toLocaleString what you want to do here? That's going to change based on the users locale - don't we want to control the format the date is in exactly?